datadog_checks_dev/datadog_checks/dev/tooling/commands/release.py

```diff
@@ -822,12 +822,19 @@ def changelog(ctx, check, version, old_version, initial, quiet, dry_run):
     short_help='Build and upload a check to PyPI'
 )
 @click.argument('check')
+@click.option('--sdist', '-s', is_flag=True)
 @click.option('--dry-run', '-n', is_flag=True)
 @click.pass_context
-def upload(ctx, check, dry_run):
+def upload(ctx, check, sdist, dry_run):
     """Release a specific check to PyPI as it is on the repo HEAD."""
-    if check not in get_valid_checks():
-        abort('Check `{}` is not an Agent-based Integration'.format(check))
+    if check in get_valid_checks():
+        check_dir = os.path.join(get_root(), check)
+    else:
+        check_dir = resolve_path(check)
+        if not dir_exists(check_dir):
+            abort('`{}` is not an Agent-based Integration or Python package'.format(check))
```

Comment:
Seems like this error would only be triggered if there isn't a directory, not if it wasn't a proper python package. I'm not sure how to reword it though

Reply:
Yeah it's tricky. Catching build errors would also be weird